apps/admin-ui/src/spa/reservations/[id]/ReservationKeylessEntrySection.tsx

```diff
@@ -0,0 +1,251 @@
+import {
+  Button,
+  ButtonSize,
+  IconAlertCircleFill,
+  IconRefresh,
+} from "hds-react";
+import { formatDate, formatTime } from "@/common/util";
+import React, { useState } from "react";
+
+import { Accordion, DataWrapper } from "@/spa/reservations/[id]/components";
+import styled, { css } from "styled-components";
+import { breakpoints } from "common";
+import { ButtonContainer } from "common/styles/util";
+import { ConfirmationDialog } from "common/src/components/ConfirmationDialog";
+import { useCheckPermission } from "@/hooks";
+
+type ReservationType = NonNullable<ReservationQuery["reservation"]>;
+
+const SummaryHorizontal = styled.div<{
+  $isRecurring?: boolean;
+}>`
+  gap: var(--spacing-s);
+  padding: var(--spacing-m);
+  background: var(--color-black-5);
+  justify-content: space-between;
+  display: grid;
+
+  ${({ $isRecurring }) =>
+    $isRecurring
+      ? css`
+          grid-template-columns: repeat(2, 1fr);
+          @media (min-width: ${breakpoints.l}) {
+            grid-template-columns: repeat(6, 1fr);
+          }
+        `
+      : css`
+          grid-template-columns: repeat(3, 1fr);
+          @media (min-width: ${breakpoints.s}) {
+            grid-template-columns: repeat(4, 1fr);
+          }
+        `};
+`;
+
+const SingleButtonContainer = styled(ButtonContainer)<{
+  $isRecurring?: boolean;
+}>`
+  ${({ $isRecurring }) =>
+    $isRecurring
+      ? css`
+          @media (max-width: ${breakpoints.l}) {
+            grid-column: 1 / span 2;
+          }
+          @media (min-width: ${breakpoints.l}) {
+            width: auto;
+          }
+        `
+      : css`
+          @media (max-width: ${breakpoints.s}) {
+            grid-column: 1 / span 3;
+          }
+          @media (min-width: ${breakpoints.s}) {
+            width: auto;
+          }
+        `};
+
+  justify-content: flex-end;
+`;
+
+export function ReservationKeylessEntry({
+  reservation,
+  onSuccess,
+}: Readonly<{
+  reservation: ReservationType;
+  onSuccess: () => void;
+}>) {
+  const { t } = useTranslation();
+
+  return (
+    <Accordion
+      id="reservation__access-type"
+      heading={t("RequestedReservation.keylessEntry")}
+      initiallyOpen={false}
+    >
+      <div>
+        <SummaryHorizontal $isRecurring={!!reservation.recurringReservation}>
+          <DataWrapper label={t("RequestedReservation.accessCodeLabel")}>
+            {reservation.pindoraInfo?.accessCode ?? "-"}
+          </DataWrapper>
+
+          <DataWrapper label={t("RequestedReservation.accessCodeStatusLabel")}>
+            {reservation.pindoraInfo?.accessCodeIsActive
+              ? t("RequestedReservation.accessCodeStatusActive")
+              : t("RequestedReservation.accessCodeStatusInactive")}
+            {reservation.pindoraInfo?.accessCodeIsActive !==
+              reservation.accessCodeShouldBeActive && <IconAlertCircleFill />}
+          </DataWrapper>
+
+          {reservation.recurringReservation && (
```

Comment:
In the recurring case, the data for `recurringReservation.pindoraInfo.accessCodeValidity` is an array which contains the begin/end times in Pindora for all the reservations in it that are supposed to have an access code; the first and last date can be looked up from it.